tests/js/spec/views/dataExport/dataDownload.spec.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import {mountWithTheme} from 'sentry-test/enzyme';
+
+import DataDownload, {DownloadStatus} from 'app/views/dataExport/dataDownload';
+
+describe('DataDownload', function() {
+  beforeEach(MockApiClient.clearMockResponses);
+  const dateExpired = new Date();
+  const mockRouteParams = {
+    orgId: 'acme',
+    dataExportId: 721,
+  };
+  const getDataExportDetails = body =>
+    MockApiClient.addMockResponse({
+      url: `/organizations/${mockRouteParams.orgId}/data-export/${mockRouteParams.dataExportId}/`,
+      body,
+    });
+
+  it('should send a request to the data export endpoint', function() {
+    const getValid = getDataExportDetails(DownloadStatus.Valid);
+    mountWithTheme(<DataDownload params={mockRouteParams} />);
+    expect(getValid).toHaveBeenCalled();
+  });
+
+  it("should render the 'Early' view when appropriate", function() {
+    const status = DownloadStatus.Early;
+    getDataExportDetails({status});
+    const wrapper = mountWithTheme(<DataDownload params={mockRouteParams} />);
+    expect(wrapper.state('download')).toEqual({status});
+    expect(wrapper.state('download').dateExpired).toBeUndefined();
+    const contentWrapper = wrapper.find('div[data-test="datadownload-wrapper"]');
+    expect(contentWrapper.children()).toHaveLength(3);
+    expect(contentWrapper.find('h3').text()).toBe("You're Early!");
+  });
+
+  it("should render the 'Expired' view when appropriate", function() {
+    const status = DownloadStatus.Expired;
+    getDataExportDetails({status});
+    const wrapper = mountWithTheme(<DataDownload params={mockRouteParams} />);
+    expect(wrapper.state('download')).toEqual({status});
+    expect(wrapper.state('download').dateExpired).toBeUndefined();
+    const contentWrapper = wrapper.find('div[data-test="datadownload-wrapper"]');
+    expect(contentWrapper.children()).toHaveLength(3);
+    expect(contentWrapper.find('h3').text()).toBe('Sorry!');
+  });
+
+  it("should render the 'Valid' view when appropriate", function() {
+    const status = DownloadStatus.Valid;
+    getDataExportDetails({status, dateExpired});
+    const wrapper = mountWithTheme(<DataDownload params={mockRouteParams} />);
+    expect(wrapper.state('download')).toEqual({dateExpired, status});
+    const contentWrapper = wrapper.find('div[data-test="datadownload-wrapper"]');
+    expect(contentWrapper.children()).toHaveLength(5);
+    expect(contentWrapper.find('h3').text()).toBe('Finally!');
+    const buttonWrapper = contentWrapper.find('a[data-test="datadownload-button"]');
```

Comment:
Instead of using a `data-test-id` attr on button, can we query via `aria-label` - this is a good practice as it also helps ensure we are being more accessible.

Reply:
Done in f387c4da13!